spanner/api/Spanner.Samples/UpdateInstanceConfigAsync.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2022 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_update_instance_config]
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.Admin.Instance.V1;
+using Google.LongRunning;
+using Google.Protobuf.WellKnownTypes;
+
+public class UpdateInstanceConfigAsyncSample
+{
+    public async Task<Operation<InstanceConfig, UpdateInstanceConfigMetadata>> UpdateInstanceConfigAsync(string projectId, string instanceConfigId)
+    {
+        InstanceAdminClient instanceAdminClient = await InstanceAdminClient.CreateAsync();
+        var instanceConfigName = new InstanceConfigName(projectId, instanceConfigId);
+
+        //Check if there is a custom config with this name.
+        var instanceConfig = await instanceAdminClient.GetInstanceConfigAsync(instanceConfigName);
```

Comment:
I think you don't need to check here. And in particular if you don't have a try catch the check may fail if the config does not exist.

Also, the purpose of the `UpdateMask` is that you don't have to have the full entity to update it. You create a new (blank) one, set the values that you want to update and set the mask accordingly. I think that shows a better use of the API/Library.

You can add a try catch to the updated operation and check for RpcException wit "NOT_FOUND" which means there was no such configuration before.